### PR TITLE
fix: 76 download button hidden for localhost

### PR DESCRIPTION
**Pull request by griffithtp**
**Type:** bugfix

The following has been addressed in the PR:

*  There is a related issue? #76 
*  Unit or Functional tests are included in the PR - YES

**Description:**
As described in the attached issue 76, download tarball files were not showing for localhost as it is not TLD.
Also, in this PR we refactor url.ts and history.ts by removing unused functions and files being called once only.

**Resolves** #76

### CODE REVIEW

**Comment by griffithtp** [comment]
Added unit test for isURL() and isEmail() however I am not entirely sure there is a purpose having `getBaseNamePath()` and `getRootPath()` in the file.
Shall we remove them if it's used only once, the other not used?

**Comment by juanpicado** [comment]
The method `getBaseNamePath` is definitely used, but about `getRootPath` I'd like to investigate what happened with the place was this being used before remove it.

**Comment by griffithtp** [comment]
> The method `getBaseNamePath` is definitely used, but about `getRootPath` I'd like to investigate what happened with the place was this being used before remove it.

`getBaseNamePath()` only used in history.ts and unless we can anticipate future use it might simpler without it and consolidate test on history.test.ts. 
```
const history = createBrowserHistory({
  basename: window.__VERDACCIO_BASENAME_UI_OPTIONS && window.__VERDACCIO_BASENAME_UI_OPTIONS.url_prefix,
});
```

Anyway this perhaps is irrelevant in this PR 😄

**Comment by juanpicado** [comment]
ahh, let's do it now, make sense, :-)

File: src/router.tsx [hunk 1]
```diff
@@ -2,12 +2,16 @@
...
 import { Router, Route, Switch } from 'react-router-dom';
+import { createBrowserHistory } from 'history';
 import { AppContextConsumer, AppStateInterface } from './App/App';
 
 import { asyncComponent } from './utils/asyncComponent';
-import history from './history';
 import Header from './components/Header';
 
+const history = createBrowserHistory({
+  basename: window.__VERDACCIO_BASENAME_UI_OPTIONS && window.__VERDACCIO_BASENAME_UI_OPTIONS.url_prefix,
```
Comment (DanielRuf) [inline src/router.tsx]: Do we need the trailing comma?

Comment (griffithtp) [reply src/router.tsx]: linting rule

**Comment by juanpicado** [comment]
Thanks @griffithtp  🚀

**Pull request merged**